beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -2216,22 +2279,59 @@ mod tests {
                     };
 
                     let mut peer_db = peer_manager.network_globals.peers.write();
-                    let peer_info = peer_db.peer_info_mut(&peer).unwrap();
+                    let peer_info = peer_db.peer_info_mut(&condition.peer_id).unwrap();
                     peer_info.set_meta_data(MetaData::V2(metadata));
                     peer_info.set_gossipsub_score(condition.gossipsub_score);
                     peer_info.add_to_score(condition.score);
 
                     for subnet in peer_info.long_lived_subnets() {
-                        peer_db.add_subscription(&peer, subnet);
+                        peer_db.add_subscription(&condition.peer_id, subnet);
                     }
                 }
 
                 // Perform the heartbeat.
                 peer_manager.heartbeat();
 
-                TestResult::from_bool(
+                // The minimum number of connected peers cannot be less than the target peer count
+                // or submitted peers.
+
+                let expected_peer_count = target_peer_count.min(peer_conditions.len());
+                // Trust peers could make this larger however.
+                let no_of_trusted_peers =
+                    peer_conditions.iter().fold(
+                        0,
+                        |acc, condition| {
+                            if condition.trusted {
+                                acc + 1
+                            } else {
+                                acc
+                            }
+                        },
+                    );
```

Comment:
what do you think about `filter(|condition|condition.trusted).count()` I think thi is optimized to a single pass over the peers and it's easier to read